lib/drivers/asanteDriver.js

```diff
@@ -1204,8 +1179,48 @@ module.exports = function (config) {
     },
 
     uploadData: function (progress, data, cb) {
+      debug('Asante UploadData!');
+      data.stage = 'uploadData';
       progress(0);
-      console.log(data);
+
+      var postrecords = [], settings = null;
+      postrecords = asanteBuildSettingsRecord(data, postrecords);
+      if (!_.isEmpty(postrecords)) {
+        settings = postrecords[0];
+      }
+      postrecords = asanteBuildBolusRecords(data, postrecords);
+      postrecords = asanteBuildWizardRecords(data, postrecords);
+      postrecords = asanteBuildBasalRecords(data, postrecords);
+      postrecords = _.sortBy(postrecords, function(d) {
+        return d.time;
+      });
+
+      var simulator = asanteSimulatorMaker.make({settings: settings});
+
+      for (var i = 0; i < postrecords.length; ++i) {
+        var datum = postrecords[i];
+        switch (datum.type) {
+          case 'basal':
+            simulator.basal(datum);
+            break;
+          case 'bolus':
+            simulator.bolus(datum);
+            break;
+          case 'settings':
+            simulator.settings(datum);
+            break;
+          case 'smbg':
+            simulator.smbg(datum);
+            break;
+          case 'wizard':
+            simulator.wizard(datum);
+            break;
+          default:
```

Comment:
Hmm, I can see your point. The signatures will not always be the same though. They only are for the simplest types. (Cf. [here](https://github.com/tidepool-org/chrome-uploader/blob/master/lib/drivers/insuletDriver.js#L1530) in the Insulet driver; I expect similar, although not exactly the same, for Asante.)

I think moving the dispatcher sounds like a good idea, but I'd prefer to wait and do it for both Insulet and Asante at the same time (for parallelism).